app/scripts/components/common/catalog/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { omit, set } from 'lodash';
+
+export enum CatalogActions {
+  CLEAR = 'clear',
+  SEARCH = 'search',
+  TAXONOMY_MULTISELECT = 'taxonomy_multiselect'
+}
+
+export type CatalogViewAction = (action: CatalogActions, value?: any) => void;
+
+export function onCatalogAction(
+  action: CatalogActions,
+  value: any,
+  taxonomies: any,
+  setSearch: (value: string) => void,
+  setTaxonomies: (value: any) => void
+) {
+  switch (action) {
+    case CatalogActions.CLEAR: {
+      setSearch('');
+      setTaxonomies({});
+      break;
+    }
+    case CatalogActions.SEARCH: {
+      setSearch(value);
+      break;
+    }
+    case CatalogActions.TAXONOMY_MULTISELECT: {
+      const { key, value: val } = value;
+
+      if (taxonomies && key in taxonomies) {
+        const taxonomyGroupValues =
+          taxonomies[key] instanceof Array
+            ? (taxonomies[key] as string[])
+            : [taxonomies[key]];
+
+        if (taxonomyGroupValues.includes(val)) {
+          const updatedValues = taxonomyGroupValues.filter((x) => x !== val);
+          updatedValues.length
+            ? setTaxonomies(set({ ...taxonomies }, key, updatedValues))
+            : setTaxonomies(omit(taxonomies, key));
+        } else {
+          taxonomyGroupValues.push(val);
+          setTaxonomies(taxonomies);
```

Comment:
Any reason not following the pattern for  `setTaxonomies(set({ ...taxonomies }, key, updatedValues))` this case? I think it is generally a better practice to avoid mutating the object.

Reply:
Re-used from before, changed.